pinot-common/src/main/java/org/apache/pinot/common/Closeables.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Arrays;
+
+
+public class Closeables {
```

Comment:
It'd be great to add some descriptions on where/where to use this class.

Reply:
+1. I think the intention is to close all Closeables even if some of them throwing exception? Another way to handle this is by using try with multiple resources:
```
    try (Closeable resource1 = closeable1; Closeable resource2 = closeable2) {
    }
```

Reply:
Added javadoc.
Yes, try with resource is a good pattern without having to worry about closing everything in finally. However, it will work only for resources that are local -- as in local to the function that are created. But for the resources that member variables of the class and are released only during the call to close(),  we can't use try with resource for them.

Reply:
> Added javadoc.
> Yes, try with resource is a good pattern without having to worry about closing everything in finally. However, it will work only for resources that are local -- as in local to the function that are created. But for the resources that member variables of the class and are released only during the call to close(), we can't use try with resource for them.

The resource doesn't need to be local. You just need to assign the member variables to local variables as in my example

Reply:
Got it. Thank you